hack/gen-versions/validate.sh

```diff
@@ -1,40 +0,0 @@
-#!/bin/bash
-# Copyright (c) 2020 Tigera, Inc. All rights reserved.
-
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#     http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-set -e
-
-git fetch origin
-
-function modded() {
-    file=$1
-    git diff --name-only HEAD..$(git merge-base HEAD origin/master) $file
```

Comment:
I think your changes are correct to remove this file and rely on dirty checks. But I just remembered why we were comparing head and wanted to explain for posterity.

This check assumes that any commit which modifies the versions.yml must also modify the rendered components, and vice versa. So it it checks that both files were _changed_. 

However, this doesn't  ensure that they're up to date. A user could, for example, make an unrelated change to each file but this test would still pass. The render-then-check-dirty you've implemented does, so 💯